docs/_posts/2015-09-30-react-render-and-top-level-api.md

```diff
@@ -0,0 +1,133 @@
+---
+title: "React.render and the Top Level React API"
+author: jimandsebastian
+---
+
+
+When you're in React's world you are just building components that fit into other components. Everything is a component. Unfortunately not everything around you is built using React. At the root of your tree you still have to write some plumbing code to connect the outer world into React.
+
+The primary API for rendering into the DOM looks like this:
+
+```js
+ReactDOM.render(reactElement, domContainerNode)
+```
+
+To update the properties of an existing component, you call render again with a new element.
+
+If you're not doing a single page app you may need to plug in to an outer component system to do cleanup work. React will not automatically clean up a tree. You need to manually call:
+
+```js
+ReactDOM.unmountComponentAtNode(domContainerNode)
+```
+
+This is important and often forgotten. Forgetting to call `unmountComponentAtNode` will cause your app to leak memory. There is no way for us to automatically detect when it is appropriate to do this work. Every system is different.
+
+It is not unique to the DOM. If you want to insert a React Native view in the middle of an existing iOS app you will hit similar issues.
+
+## Helpers
+
+If you have multiple React roots, or a single root that gets deleted over time, we recommend that you always create your own wrapper API. These will all look slightly different depending on what your outer system looks like. For example, at Facebook we have a system that automatically ties into our page transition router to automatically call `unmountComponentAtNode`.
+
+It is an anti-pattern to call `ReactDOM.render` directly.
```

Comment:
Sebastian wanted a "memorable quote" that would have impact in people's minds.  @sebmarkbage do you still like this line or should we remove it?  Perhaps:
 > Rather than calling `ReactDOM.render()` directly, consider writing/using a library that will manage mounting and unmounting within your application.